### ISSUE TITLE
Incorrect report of force_unwrap violation in string (from Sourcery code base)

### ISSUE THREAD

**Issue by Liquidsoul** [realm/SwiftLint#1614]
Hi,

While fixing the swiftlint warnings in [Sourcery](https://github.com/krzysztofzablocki/Sourcery/pull/351), I noticed that there were some warnings about force unwrapping that should not be there:

> AutoEquatable.swift:133:27: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
> AutoHashable.swift:70:27: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
> AutoHashable.swift:121:27: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)

They all involve the same code:

```swift
func greeting(for name: String) -> String {
    return "Hi \(name)!"
}
```

I tried to isolate a triggering example and it seems that you need to include the whole method:

```swift
"func someFunction(for name: String) -> String {\n    return \"Hi \\(name)!\n\"}"
```

I traced the condition of the violation and it seems that it is because of its [kind](https://github.com/realm/SwiftLint/blob/f8ca8640c38c5c110547e4d6bcab122f903d9821/Source/SwiftLintFramework/Rules/ForceUnwrappingRule.swift#L169) while checking the structure.
Unfortunately, I do not understand what this means so I cannot offer a fix through a PR 😓 

Hope this helps!

**Comment by marcelofabri** [comment]
#1584 should fix this

**Comment by marcelofabri** [comment]
Fixed in #1620.
